### PR TITLE
Update to Angular 1.6

### PR DESCRIPTION
**Pull request by dgsmith2**
Updating code and tests to address #6360, making angular-ui/bootstrap compatible with Angular 1.6.  These are **BREAKING CHANGES**, incompatible with Angular 1.5.

### CODE REVIEW

**Comment by dgsmith2** [comment]
@wesleycho, what's the timeline for this being merged and a new release being cut?  We are still using Bower for dependencies, so I'm in search of the path of least resistance for myself to get this code into our app.

File: package.json [hunk 1]
```diff
@@ -30,9 +30,9 @@
     "url": "https://github.com/angular-ui/bootstrap.git"
   },
   "devDependencies": {
-    "angular": "1.5.8",
-    "angular-mocks": "1.5.8",
-    "angular-sanitize": "1.5.8",
+    "angular": "1.6.0",
+    "angular-mocks": "1.6.0",
+    "angular-sanitize": "1.6.0",
```
Comment (wesleycho) [inline package.json]: These can all probably be bumped to 1.6.1

File: src/datepicker/datepicker.js [hunk 2]
```diff
@@ -161,14 +161,24 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
...
-      datepickerConfig.ngModelOptions;
+
+    // grab any timezone value based on precedence
+    var timezone = ngModelCtrl_.$options.getOption('timezone') ||
+      ($scope.datepickerOptions.ngModelOptions ? $scope.datepickerOptions.ngModelOptions.timezone : null) ||
+      (datepickerConfig.ngModelOptions ? datepickerConfig.ngModelOptions.timezone : null);
+
+    // need to start with existing ModelOptions but also give it greatest precedence minus its default of null timezone
+    ngModelOptions = ngModelCtrl_.$options
+      .createChild(datepickerConfig.ngModelOptions)
```
Comment (wesleycho) [inline src/datepicker/datepicker.js]: This breaks compatibility with 1.5 I believe - we cannot have that for a minor version bump. Is there no way to made this compatible with both versions?

File: src/datepicker/datepicker.js [hunk 3]
```diff
@@ -161,14 +161,24 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
...
+      .createChild($scope.datepickerOptions.ngModelOptions)
+      .createChild(ngModelCtrl_.$options)
+      .createChild({timezone: timezone});
+
     if ($scope.datepickerOptions.initDate) {
-      self.activeDate = dateParser.fromTimezone($scope.datepickerOptions.initDate, ngModelOptions.timezone) || new Date();
+      self.activeDate = dateParser.fromTimezone($scope.datepickerOptions.initDate, ngModelOptions.getOption('timezone')) || new Date();
```
Comment (wesleycho) [inline src/datepicker/datepicker.js]: Ditto with `getOption`

**Comment by wesleycho** [comment]
This LGTM minus the compatibility break - if this is to be merged earlier, then we absolutely cannot break compatibility yet. A mechanism for handling the ngModel.$options differences needs to be created to handle differences in 1.5 and 1.6.

File: package.json (hunk 1 above)
Comment (dgsmith2) [inline package.json]: done

File: src/datepicker/datepicker.js (hunk 2 above)
Comment (dgsmith2) [inline src/datepicker/datepicker.js]: done

File: src/datepicker/datepicker.js (hunk 3 above)
Comment (dgsmith2) [inline src/datepicker/datepicker.js]: done

**Comment by dgsmith2** [comment]
@wesleycho I'm pretty confident things are good to do.  As you can see, the tests past.  Locally, I also reset my version of Angular back to 1.5 and re-ran the tests.  All tests in regards to $options still passed.  The only failure, which was expected, was in regards to 1.5 adding the extra span for transclusion--I was surprised the other updated transclusion tests didn't fail.

**Comment by wesleycho** [comment]
Thanks - I will take another pass over the weekend (sleep deprived currently) and if things are good, I will merge & cut a release then.

**Pull request closed without merging**